### PR TITLE
boom

### PR DESCRIPTION
**Pull request by Cashmaney**

### CODE REVIEW

**Comment by toml01** [comment]
Give that guy a medal! 🏅

File: contract/src/contract.rs [hunk 1]
```diff
@@ -134,15 +131,64 @@ pub enum HandleMsg {
...
+            let contract_address = deps.api.human_address(&env.contract.address)?;
+
+            let mut table: Table =
+                serde_json::from_slice(&deps.storage.get(b"table").unwrap()).unwrap();
+
+            if table.player_a_wallet == 0 && player_name == table.player_b.unwrap() {
+                Ok(winner_winner_chicken_dinner(contract_address, player_name))
+            } else if table.player_b_wallet == 0 && player_name == table.player_a.unwrap() {
+                Ok(winner_winner_chicken_dinner(contract_address, player_name))
+            }
```
Comment (reuvenpo) [inline contract/src/contract.rs]: `Some(player_name) == table.player_b` instead of unwrap

File: contract/src/contract.rs [hunk 2]
```diff
@@ -134,15 +131,64 @@ pub enum HandleMsg {
...
+            };
+
+            if env.message.sent_funds.len() == 0 {
+                return Err(generic_err("SHOW ME THE MONEY"));
+            } else {
+                let deposit = env.message.sent_funds[0].clone();
+
+                if deposit.ne(&min_deposit) {
+                    return Err(generic_err("WRONG MONEY"));
+                }
```
Comment (reuvenpo) [inline contract/src/contract.rs]: `if deposit != min_deposit`

File: contract/src/contract.rs [hunk 3]
```diff
@@ -134,15 +131,64 @@ pub enum HandleMsg {
...
+
+            let min_deposit: Coin = Coin{
+                denom: "uscrt".to_string(),
+                amount: Uint128(100_000_000),
+            };
+
+            if env.message.sent_funds.len() == 0 {
+                return Err(generic_err("SHOW ME THE MONEY"));
+            } else {
+                let deposit = env.message.sent_funds[0].clone();
```
Comment (reuvenpo) [inline contract/src/contract.rs]: you don't want to sum up all the sent funds?
also you need to check that they're all `uscrt`

File: contract/src/contract.rs [hunk 4]
```diff
@@ -134,15 +131,64 @@ pub enum HandleMsg {
...
     msg: HandleMsg,
 ) -> HandleResult {
     match msg {
+        HandleMsg::Withdraw {} => {
+
+            let player_name = deps.api.human_address(&env.message.sender)?;
+            let contract_address = deps.api.human_address(&env.contract.address)?;
+
+            let mut table: Table =
+                serde_json::from_slice(&deps.storage.get(b"table").unwrap()).unwrap();
```
Comment (reuvenpo) [inline contract/src/contract.rs:220]: I believe that you can use `?` for the `storage.get` and `serde_json::from_slice` can be replaced with `cosmwasm_std::from_slice` which also can be used with `?` here.

**Pull request merged**